pytrickle/client.py

```diff
@@ -269,11 +308,20 @@ async def _control_loop(self):
     async def _send_data_loop(self):
         """Send data to the server every 333ms, batching all available items."""
         try:
-            while not self.stop_event.is_set():
-                # Wait for send_data_interval or until stop event is set
+            while not self.stop_event.is_set() and not self.error_event.is_set():
+                # Wait for send_data_interval or until stop/error event is set
                 try:
-                    await asyncio.wait_for(self.stop_event.wait(), timeout=self.send_data_interval)
-                    break  # Stop event was set, exit loop
+                    done, pending = await asyncio.wait(
```

Comment:
WDYT about just waiting for stop_event here?

Will error_event cause channels to close down before this could execute one last time?

Reply:
Good catch!

Reply:
I tests and it works great, no issues found in stopping the publisher. Ran a high-rate publisher test and got ~120 msg/s which was expected. One message did fail, but did not catch why, likely due to message size limit from batching

Reply:
We could possibly improve this by allowing `_wait_for_interval` to raise it's own error